clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -1960,9 +1961,11 @@ void CodeGenModule::GenOpenCLArgMetadata(llvm::Function *Fn,
                     llvm::MDNode::get(VMContext, argBaseTypeNames));
     Fn->setMetadata("kernel_arg_type_qual",
                     llvm::MDNode::get(VMContext, argTypeQuals));
+    // FIXME: What is the purpose of this metadata for ESIMD? Can we
+    // reuse kernel_arg_exclusive_ptr or kernel_arg_runtime_aligned ?
```

Comment:
This FIXME is against a now-non-existent metadata for SIMD.

Reply:
Can you please explain? There no ESIMD changes in this patch. I just reused a variable to generate the ESIMD metadata. We had 2 different variables collecting the same information

Reply:
To clarify, prior to this patch, `argSYCLKernelRuntimeAligned `and `argESIMDAccPtrs` collected the same information. The only difference being, `argESIMDAccPtrs` was only populated in ESIMD case. We then used these variables to generate corresponding metadata. I just renamed the variables to use a common one.

Reply:
Yes I believe so. `argSYCLAccessorPtrs` is populated with required information always. It is used to generate metadata for both cases